text/3729-sized-hierarchy.md

```diff
@@ -0,0 +1,1546 @@
+  escape-the-sized-hierarchy part with the Custom DST machinery.
+    - e.g. `DynSized` trait with `Metadata` associated types and `size_of_val`
+      and `align_of_val` methods, or a `!Sized` pseudo-trait that you could implement.
+    - Given the acceptance of [rfcs#2580][rfc_pointer_metadata_vtable], Rust
+      doesn't seem to be trending in this direction, as the `Metadata` part of this is
+      now part of a separate `Pointee` trait.
+- Most early `DynSized` trait proposals (independent or as part of Custom DSTs)
+  would make `DynSized` a default bound mirroring `Sized`, and consequently had a
+  relaxed form `?DynSized`.
+    - Later proposals were more aware of the language team's resistance towards
+      adding new relaxed bounds and tried to avoid this.
+- Backwards compatibility concerns were the overriding reason for the rejection
+  of previous `DynSized` proposals.
+    - These can be sidestepped by avoiding having a relaxed form and by relying
+      on being a supertrait of `Sized`.
+
+The [Rationale and Alternatives](#rationale-and-alternatives) section provides
+rationale for some of the decisions made in this RFC and references the prior
+art above when those proposals made different decisions.
+
+No previous proposal captures the specific part of the design space that this
+proposal attempts to, but these proposals are the closest matches for parts of
+this proposal:
+
+- [Pre-eRFC: Let's fix DSTs][pre_erfc_fix_dsts] was the only other proposal
+  removing `Sized` bounds when a bound for another sized trait (only `DynSized`
+  in that pre-eRFC's case) was present.
+    - However, this proposal had `size_of_val` methods in its `DynSized` trait and
+      proposed a bunch of other things necessary for Custom DSTs.
+- [rfcs#2310: DynSized without ?DynSized][rfc_dynsized_without_dynsized] was
+  proposed at a similar time and was similarly focused only on making `Sized` more
+  flexible, but had a bunch of machinery for avoiding backwards incompatibility
+  that this RFC believes is unnecessary. Like this proposal, it avoided making
+  `DynSized` a default bound and avoided having a relaxed form of it.
+    - However, this proposal didn't suggest removing default `Sized` bounds in
+      the presence of other size trait bounds.
+- [Sized, DynSized, and Unsized][blog_dynsized_unsized] is very similar and a
+  major inspiration for this proposal. It has everything this proposal has except
+  for the const size traits and all the additional context an RFC needs.
+
+Some prior art referenced [rust#21974][issue_regions_too_simplistic] as a limitation
+of the type system which can result in new implicit bounds or implicit supertraits
+being infeasible for implementation reasons, but [it is believed that this is no
+longer relevant][zulip_issue_regions_too_simplistic].
+
+# Unresolved questions
+[unresolved-questions]: #unresolved-questions
+
+None currently.
+
+# Future possibilities
+[future-possibilities]: #future-possibilities
```

Comment:
I am not smart/informed enough to tell for sure, but I _feel_ like if we want to change the whole default `?Sized` thing, and necessarily deal with the breakage/fallout from that, we might want to consider adding `Move` at the same time?

Anyhow, just a consideration, excellent RFC, full support from my side!

Reply:
Be still my heart. 

I guess this could just be the Baader-Meinhof phenomenon at work, but I feel like `Move` is coming up a lot recently. Do you know if anybody has put together recent summary of the "state of `Move`" — i.e. plausibility of ever having it even theoretically, would it require an edition, scale of the challenge, etc.?

Reply:
I haven't read enough of the context of `Move` to know whether it makes sense to incorporate into this proposals, but I assume it doesn't. If someone wants to go and summarise the history of those proposals and make a compelling case for how it could be integrated with this work then I'd be open to add it as a alternative that the language team could consider.

Reply:
This RFC is changing the `Sized` axis of default bounds, an axis where we already have an existing default bound. I don't see how that would make it any easier to add an entirely new axis of default bounds.